File: 백준/Silver/9012. 괄호/괄호.py
for _ in range(int(input())):
    parenthesis = input()
    stack = []
    for char in parenthesis:
        if char == ")" and stack:
            stack.pop()
        elif char == ")" and not stack:
            stack.append(")")
            break
        else:
            stack.append(char)
    if stack:
        print("NO")
    else:
        print("YES")

File: 백준/Silver/9012. 괄호/괄호.swift
let n = Int(readLine()!)!
for _ in 1...n {
    let chars = readLine()!
    let arr = chars.map { String($0) }
    var stack: [String] = []
    for item in arr {
        if item == "(" {
            stack.append(item)
        } else if item == ")" {
            if stack.isEmpty {
                stack.append(item)
                break
            } else {
                stack.popLast()
            }
        }
    }
    if stack.isEmpty {
        print("YES")
    } else {
        print("NO")
    }
}